external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -2151,23 +2305,55 @@ impl crate::shared::ast_debug::AstDebug for ExpDotted_ {
     }
 }
 
+fn annotate_<S: ToString, F: FnOnce() -> S>(
+    context: &mut Context,
+    msg: F,
+    annot_loc: Loc,
+    e: Box<T::Exp>,
+    ty_annot: Type,
+) -> (Type, T::UnannotatedExp_) {
+    let rhs = core::instantiate(context, ty_annot);
+    subtype(context, annot_loc, msg, e.ty.clone(), rhs.clone());
+    let e_ = match core::unfold_type(&context.subst, rhs.clone()).value {
+        Type_::Fun(_, _) => e.exp.value,
```

Comment:
Are we doing this because the annotation is going away anyway and we don't want to carry it into macro expansion and have to find and get rid of it there?

Reply:
We are doing this because I missed removing this from when you used to be able to write `foo!((|x| x: |u64| -> u64))` now you have to write `foo!(|x: u64| -> u64 { x })` 

So while this case should not get hit, I'll clean it up